packages/lodestar/src/api/impl/validator/validator.ts

```diff
@@ -119,6 +116,8 @@ export class ValidatorApi implements IValidatorApi {
 
   public async publishBlock(signedBlock: SignedBeaconBlock): Promise<void> {
     await this.checkSyncStatus();
+    await waitForBlockSlot(this.config, this.chain.getGenesisTime(), signedBlock.message.slot);
```

Comment:
maybe it would be better to have waitForSlot method on chain clock?

It would be nice if we would use chain clock everywhere to get time as it would make it easier to implement some time sync strategy in future

Reply:
Yeah, I can add add it to this PR.